sqlx-core/src/connection.rs

```diff
@@ -49,6 +50,16 @@ pub trait Connection: Send {
     where
         Self: Sized;
 
+    /// Begin a new transaction with a custom statement.
+    ///
+    /// Returns a [`Transaction`] for controlling and tracking the new transaction.
```

Comment:
This should document that it returns an error if the given statement does not create a transaction, or the connection is already in a transaction.